### PR TITLE
drivers/timer: stm32: Convert configuration from Kconfig to device tree

### PR DESCRIPTION
**Pull request by erwango**
Use device tree to configure lptim driver with following benefits:
- Allow usage of instances other than lptim1 if needed
- Simplify driver code, specially lptim source clock selection (thanks to use of dt domain clock)

Make some related clean up in board description files.

### CODE REVIEW

**Comment by erwango** [comment]
CI failed due to `tests/net/` failing on mps2 target:
```
tests/net/socket/tcp/net.socket.tcp (mps2_an385:tests/net/socket/tcp/net.socket.tcp) failed
```

File: drivers/timer/stm32_lptim_timer.c [hunk 1]
```diff
@@ -27,6 +27,9 @@
 
 #define LPTIM (LPTIM_TypeDef *) DT_INST_REG_ADDR(0)
 
+static const struct stm32_pclken lptim_clk[] = STM32_DT_INST_CLOCKS(0);
+static const struct device *clk_ctrl = DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE);
```
Comment (gmarull) [inline drivers/timer/stm32_lptim_timer.c:46]: nit: check for readiness if https://github.com/zephyrproject-rtos/zephyr/pull/48787 goes first

**Review by gmarull — DISMISSED** [review]

File: drivers/timer/stm32_lptim_timer.c [hunk 2]
```diff
@@ -281,44 +291,26 @@ static int sys_clock_driver_init(const struct device *dev)
...
-	/* enable LSE clock */
-	LL_RCC_LSE_DisableBypass();
-#ifdef RCC_BDCR_LSEDRV_Pos
-	LL_RCC_LSE_SetDriveCapability(STM32_LSE_DRIVING << RCC_BDCR_LSEDRV_Pos);
-#endif
-	LL_RCC_LSE_Enable();
-	while (!LL_RCC_LSE_IsReady()) {
-		/* Wait for LSE ready */
+	/* A 32bit value cannot exceed 0xFFFFFFFF/lptim_time_base counting cycles.
+	 * This is for example about of 65000 x 2000ms when clocked by LSI
```
Comment (ABOSTM) [inline drivers/timer/stm32_lptim_timer.c:305]: The original comment should not be moved,
And to explain the 0xFA00 value:
`When clocked by LSI @32K:   32000Hz x 2s = 65000 = 0xFA00`

**Review by ABOSTM — CHANGES_REQUESTED** [review]

File: boards/arm/nucleo_wb55rg/nucleo_wb55rg.dts [hunk 3]
```diff
@@ -179,6 +179,8 @@
 };
 
 &lptim1 {
+	clocks = <&rcc STM32_CLOCK_BUS_APB1 0x80000000>,
+		 <&rcc STM32_SRC_LSE LPTIM1_SEL(1)>;
```
Comment (FRASTM) [inline boards/arm/nucleo_wb55rg/nucleo_wb55rg.dts:183]: STM32_SRC_LSE LPTIM1_SEL(**3**)

**Review by FRASTM — DISMISSED** [review]

File: drivers/timer/stm32_lptim_timer.c (hunk 1 above)
Comment (erwango) [reply drivers/timer/stm32_lptim_timer.c:46]: Thanks, done

**Review by ABOSTM — APPROVED** [review]
LGTM

**Review by FRASTM — APPROVED** [review]

**Pull request merged**